airflow/contrib/hooks/gcp_dataflow_hook.py

```diff
@@ -216,7 +620,12 @@ def _set_variables(variables):
         return variables
 
     def start_java_dataflow(self, job_name, variables, dataflow, job_class=None,
-                            append_job_name=True):
+        append_job_name=True, cancel_existing_job=False, context=None):
+        self._validate_params(cancel_existing_job, append_job_name)
+        if not append_job_name and context:
```

Comment:
Why do we have "context" as part of the condition?

Reply:
we are extracting the DAG's execution date from the context. Hence i added a check that it should not be empty.
But now that i look at it, i think it's better to let the operator fail if context is somehow missing. I will remove it